pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -179,27 +181,86 @@ func (h *handler) getChartsToInstall() []*chart.Definition {
 			Values: func() map[string]interface{} {
 				values := map[string]interface{}{}
 				// add priority class value
-				if priorityClassName, err := h.chartsConfig.GetGlobalValue(chart.PriorityClassKey); err != nil {
-					if !chart.IsNotFoundError(err) {
-						logrus.Warnf("Failed to get rancher %s for 'rancher-provisioning-capi': %s", chart.PriorityClassKey, err.Error())
-					}
-				} else {
-					values[priorityClassKey] = priorityClassName
-				}
-
+				h.setPriorityClass(values, chart.ProvisioningCAPIChartName)
 				// get custom values for the rancher-provisioning-capi
-				configMapValues, err := h.chartsConfig.GetChartValues(chart.ProvisioningCAPIChartName)
-				if err != nil && !chart.IsNotFoundError(err) {
-					logrus.Warnf("Failed to get rancher-provisioning-capi %s", err.Error())
-				}
-
+				configMapValues := h.getChartValues(chart.ProvisioningCAPIChartName)
 				return data.MergeMaps(values, configMapValues)
 			},
 			Enabled:         func() bool { return true },
 			Uninstall:       !features.EmbeddedClusterAPI.Enabled(),
 			RemoveNamespace: !features.EmbeddedClusterAPI.Enabled(),
 		},
+		{
+			ReleaseNamespace:    namespace.System,
+			ChartName:           chart.SystemUpgradeControllerChartName,
+			ExactVersionSetting: settings.SystemUpgradeControllerChartVersion,
+			Values: func() map[string]interface{} {
+				values := map[string]interface{}{}
+				// add priority class value
+				h.setPriorityClass(values, chart.SystemUpgradeControllerChartName)
+				// get custom values for system-upgrade-controller
+				configMapValues := h.getChartValues(chart.SystemUpgradeControllerChartName)
+				return data.MergeMaps(values, configMapValues)
+			},
+			Enabled: func() bool {
+				toEnable := false
+				suc, err := h.deployment.Get(namespace.System, sucDeploymentName)
```

Comment:
We do not handle the case where `err` is not nil and _not_ equal to `IsNotFound`. Is that okay?

Reply:
I think the case is covered:  
if the `err` is not nil and not equal to `IsNotFound`, then the `suc` will be nil and therefore `toEnable` will be false which is its default value. 
Does it sound right to you?

Reply:
Yeah that makes sense, should we have a log message here though? I'm not sure familiar on the circumstances where this could error out, so if we don't feel that logging the error is important then this comment can be disregarded.

Reply:
A debug-level message is added.